### ISSUE TITLE
maximizing or fullscreen mode cause client focusing problems

### ISSUE THREAD

**Issue by franzhuang** [awesomeWM/awesome#3468]
Output of `awesome --version`: 

awesome v4.3 (Too long)
 • Compiled against Lua 5.4.2 (running with 0.9.2)
 • D-Bus support: ✔
 • execinfo support: ✔
 • xcb-randr version: 1.6
 • LGI version: /usr/share/lua/5.4/lgi/version.lua

**How to reproduce the issue:**

I have these configuration:

```
    -- By-direction client focus
    awful.key({ modkey }, "j",
        function()
            awful.client.focus.global_bydirection("down")
            if client.focus then client.focus:raise() end
        end,
        {description = "focus down", group = "client"}),
    awful.key({ modkey }, "k",
        function()
            awful.client.focus.global_bydirection("up")
            if client.focus then client.focus:raise() end
        end,
        {description = "focus up", group = "client"}),
    awful.key({ modkey }, "h",
        function()
            awful.client.focus.global_bydirection("left")
            if client.focus then client.focus:raise() end
        end,
        {description = "focus left", group = "client"}),
    awful.key({ modkey }, "l",
        function()
            awful.client.focus.global_bydirection("right")
            if client.focus then client.focus:raise() end
        end,
        {description = "focus right", group = "client"}),
```
When I maximized one of the client, I can't focus on all of my clients but previous client using modkey+hjkl.
The same thing happens when I am in fullscreen mode.

**Actual result:**

_REPLACE with what happened_

**Expected result:**

If I can modkey+hjkl to all of clients even if I am in fullscreen mode or maximizing some of clients, it will be nice.

**Comment by franzhuang** [comment]
Indeed, but I think it might be convenient to use up or down ( jk ) to refer to both direction and index.
And I found that even "focus by index" not going through all the opened clients.

**Comment by actionless** [comment]
1) it indeed seems like the current behavior of focus.* actions while being focused on fullscreen client laying around in undefined behavior area

2) from the point of logic as @Aire-One says it would make sense to not handle those at all

3) **however** i myself using this undocumented 'feature' quite often - which leads us to the point, what we prolly need to sit down and design that behavior to be defined - at least in words - at from there to see how the current implementation differs from that design
